app/models/user.rb

```diff
@@ -41,14 +40,30 @@ def forget
     update_attribute :remember_digest, nil
   end
 
-  def authenticated? remember_token
-    return false unless remember_digest
-    BCrypt::Password.new(remember_digest).is_password? remember_token
+  def authenticated? attribute, token
+    digest = send "#{attribute}_digest"
+    return false unless digest
+
+    BCrypt::Password.new(digest).is_password? token
+end
+
+  def activate
+   update_attribute :activated, true
+   update_attribute :activated_at, Time.zone.now
```

Comment:
```suggestion
   update_attributes activated: true, activated_at: Time.zone.now
```

Reply:
this oneline code cannot activate the user, i have tried just now

Reply:
please use: `update activated: true, activated_at: Time.zone.now`

Reply:
Mr Manh, it does not work also, even if I change to update_attributes

Reply:
try to restart server ?

Reply:
tried,still canoot,has re-login issue